pytket/extensions/qujax/qujax_convert.py

```diff
@@ -136,53 +80,60 @@ def tk_to_qujax_args_symbolic(
+            param_inds_seq.append(jnp.arange(param_index, param_index + n_params))
+            param_index += n_params
 
     return gate_name_seq, qubit_inds_seq, param_inds_seq, circuit.n_qubits
 
 
-def tk_to_qujax_symbolic(
-    circuit: Circuit, symbol_map: Optional[dict] = None
+def tk_to_qujax(
+    circuit: Circuit, symbol_map: Optional[Union[bool, dict]] = False
 ) -> qujax.UnionCallableOptionalArray:
     """
-    Converts a pytket circuit with symbolics parameters and a symbolic parameter map
-    into a function that maps circuit parameters to a statetensor.
-    Assumes all circuit gates can be found in qujax.gates.
-    Note that the behaviour of tk_to_qujax_symbolic(circuit)
-    is different to tk_to_qujax(circuit),
-    tk_to_qujax_symbolic will look for parameters in circuit.free_symbols()
-    and if there are none it will assume that none of the gates require parameters.
-    On the other hand, tk_to_qujax will work out which gates are parameterised
-    based on e.g. circuit.get_commands()[0].op.params
+    Converts a pytket circuit into a function that maps circuit parameters
+    to a statetensor. Assumes all circuit gates can be found in qujax.gates.
+    If symbol_map is False, parameter arguments of created function are ordered as in
+    circuit.get_commands() - pytket automatically reorders some gates.
+    Consider Barriers. If symbol_map is True or provided, look for parameters in
+    circuit.free_symbols() instead. Remaining parameters will be treated as constants.
+    If there are no symbolic params,
+    will assume that none of the gates require parameters.
 
     :param circuit: Circuit to be converted.
     :type circuit: pytket.Circuit
-    :param symbol_map: dict that maps elements of circuit.free_symbols() (sympy)
-        to parameter indices.
-    :type symbol_map: Optional[dict]
-    :return: Function which maps parameters
-        (and optional statetensor_in) to a statetensor.
+    :param symbol_map:
+        if False, treats all tket parameters as qujax arguments
+        if True, symbolic parameters are arguments and numerical parameters as constants
+        if dict, maps tket parameters to qujax arguments following the dict provided
+    :type symbol_map: Optional[Union[bool, dict]]
+    :return: Function which maps parameters (and optional statetensor_in)
+        to a statetensor.
         If the circuit has no parameters, the resulting function
         will only take the optional statetensor_in as an argument.
     :rtype: Callable[[jnp.ndarray, jnp.ndarray = None], jnp.ndarray]
         or Callable[[jnp.ndarray = None], jnp.ndarray]
         if no parameters found in circuit
     """
-    return qujax.get_params_to_statetensor_func(
-        *tk_to_qujax_args_symbolic(circuit, symbol_map)
-    )
+
+    return qujax.get_params_to_statetensor_func(*tk_to_qujax_args(circuit, symbol_map))
 
 
 def print_circuit(
     circuit: Circuit,
+    symbol_map: Optional[Union[bool, dict]] = False,
```

Comment:
I think we only need the `Optional` if its default value is `None` ?

Either that or we should add `Optional` to the extra arguments in `print_circuit`

Reply:
FIxed!